main/Select/Select.tsx

```diff
@@ -0,0 +1,275 @@
+import {
+  Animated,
+  Easing,
+  StyleProp,
+  TextStyle,
+} from 'react-native';
+import React, { Fragment, ReactNode } from 'react';
+
+import Arrow from './Arrow';
+import LoadingIndicator from '../LoadingIndicator';
+import { THEME } from './theme';
+import styled from 'styled-components/native';
+
+interface Props {
+  testID?: string; // Id for tests
+  open?: boolean; // Select is toggled?
+  loading?: boolean; // is Loading?
+  disabled?: boolean; // is Disabled?
+  showArrow?: boolean; // show Right Arrow?
+  dark?: boolean; // dark mode
+  bordered?: boolean; // with round borders?
+  activeOpacity?: number; // set the opacity of selected value
+  listHeight?: number; // fix the height of dropdown list
+  placeholder?: string | number; // placeholder value
+  value?: string | number; // current selected value
+  defaultValue?: string | number; // initial value instead of placeholder
+  onSelect?: (param: string | number | undefined) => any;
+  onOpen?: (param: boolean) => any;
+  prefixIcon?: ReactNode; // Icon on the left of placeholder
+  suffixIcon?: ReactNode; // Icon on the right of placeholder
+  customLoader?: ReactNode; // custom loader while loading is true
+  textStyle?: StyleProp<TextStyle>;
+  style?: React.CSSProperties;
+  children?: ReactNode;
+}
+
+const Container = styled.View`
+  width: 100%;
+  height: 100%;
+`;
+const RootWrapper = styled.TouchableOpacity<{
+  dark: boolean;
+  disabled: boolean;
+  bordered: boolean;
+}>`
+  width: 100%;
+  height: 100%;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding: 14px 12px;
+  background-color: ${({ dark, disabled }): string =>
+    dark
+      ? disabled
+        ? THEME.DARK.DISABLED.backgroundColor
+        : THEME.DARK.backgroundColor
+      : disabled
+        ? THEME.LIGHT.DISABLED.backgroundColor
+        : THEME.LIGHT.backgroundColor};
+  border-radius: ${({ bordered }): string => (bordered ? '5px' : '0')};
+  border-color: ${({ dark, disabled }): string =>
+    dark
+      ? disabled
+        ? THEME.DARK.DISABLED.borderColor
+        : THEME.DARK.borderColor
+      : disabled
+        ? THEME.LIGHT.DISABLED.borderColor
+        : THEME.LIGHT.borderColor};
+  box-shadow: 0px 2px 4px rgba(212, 210, 212, 0.8);
+  opacity: 0.7;
+`;
+const SelectListModal = styled.Modal<{
+  visible: boolean;
+  dark: boolean;
+  bordered: boolean;
+}>`
+  width: 100%;
+  border-color: transparent;
+  background-color: ${({ dark }): string =>
+    dark
+      ? THEME.DARK.INVERTED.backgroundColor
+      : THEME.LIGHT.INVERTED.backgroundColor};
```

Comment:
If this is your intention,
What about providing general `ThemeType` for your component?

```
const light = {
  backgroundColor: '#fff',
  ...
}

export type SelectTheme = typeof light;

const dark: SelectTheme = {
 ...
}

```

Reply:
I allowed the possibility to add user custom styles. For now, I'll leave the `ThemeType` providing. I will refactor this soon.